spot_rl_experiments/spot_rl/utils/heuristic_nav.py

```diff
@@ -0,0 +1,405 @@
+                if len(indices_in_x_limits) > 0:
+                    zs = zs[indices_in_x_limits]
+                    corners_yxs = corners_yxs[indices_in_x_limits]
+                    # point_3ds = point_3ds[indices_in_x_limits]
+                    mu, std = st.norm.fit(zs)
+                    combined_score = 1 * np.absolute(
+                        zs - mu
+                    )  # + 2*distance_from_object_to_corners
+
+                    # print("Gaussian Depth Distribution for the corners ", mu, std)
+                    min_arg = np.argmin(combined_score)
+                    best_corner_yx = corners_yxs[min_arg]
+                    best_offseted_z = zs[min_arg]
+                    return True, best_offseted_z, best_corner_yx, corners_yxs
+                else:
+                    return False, "couldnot find best corner within x limit", None, None
+            else:
+                return False, "couldnot find best corner below y limit", None, None
+        else:
+            return False, "couldnot find best corner closer than z", None, None
+    else:
+        return False, "couldnot find best non zero depth corner", None, None
+
+
+def convert_point_from_local_to_global_nav_target(
+    point_in_local_3d: np.ndarray,
+    direction_local: np.ndarray,
+    spot: Spot,
+    vision_T_hand: mn.Matrix4,
+    body_T_hand: mn.Matrix4,
+):
+
+    # point_in_body:mn.Vector3 = body_T_hand.transform_point(mn.Vector3(*point_in_local_3d))
+
+    point_in_global_3d: mn.Vector3 = vision_T_hand.transform_point(
+        mn.Vector3(*point_in_local_3d)
+    )
+    point_in_global_3d = np.array(
+        [point_in_global_3d.x, point_in_global_3d.y, point_in_global_3d.z]
+    )
+    # print(f"Point in global 3d, {point_in_global_3d}")
+    theta = np.arctan(point_in_global_3d[1] / point_in_global_3d[0])
+    # print(f"theta before transform {np.degrees(theta)}")
+    global_x, global_y, transformed_theta = spot.xy_yaw_global_to_home(
+        point_in_global_3d[0], point_in_global_3d[1], theta
+    )
+    # Priyam's logic
+    curr_x, curr_y, curr_yaw = spot.get_xy_yaw()
+    delta_vec = np.array([global_x, global_y]) - np.array([curr_x, curr_y])
+    # norm_delta_vec = np.linalg.norm(delta_vec)
+    theta = np.arctan2(delta_vec[1], delta_vec[0])
+    print(
+        f"transformed theta {np.degrees(transformed_theta)}, new theta {np.degrees(theta)}"
+    )
+    return (global_x, global_y), theta
+
+
+def pull_back_point_along_theta_by_offset(
+    x: float, y: float, theta: float, offset: float = 0.5
+):
+    """
+    Pulls back the x,y along theta direction for static offset in meters
+    """
+    x, y = x - offset * np.cos(theta), y - offset * np.sin(theta)
+    return (x, y)
+
+
+def push_forward_point_along_theta_by_offset(
+    x: float, y: float, theta: float, offset: float = 0.5
+):
+    x, y = x + offset * np.cos(theta), y + offset * np.sin(theta)
+    return (x, y)
+
+
+def get_me_arguments_for_image_search_fn(spot: Spot, angle):
```

Comment:
Is "me" an abbreviation for something here?

Reply:
changed the name